src/scripts/Update.ts

```diff
@@ -1513,6 +1518,253 @@ class Update implements Saveable {
+                [-869.35, 869.32],
+                [-869.36, 869.38],
+                [-869.37, 869.33],
+                [-869.38, 869.35],
+                [-869.41, 869.46],
+                [-869.42, 869.47],
+                [-869.43, 869.44],
+                [-869.44, 869.41],
+                [-869.45, 869.42],
+                [-869.46, 869.48],
+                [-869.47, 869.43],
+                [-869.48, 869.45],
+                [-869.51, 869.56],
+                [-869.52, 869.57],
+                [-869.53, 869.54],
+                [-869.54, 869.51],
+                [-869.55, 869.52],
+                [-869.56, 869.58],
+                [-869.57, 869.53],
+                [-869.58, 869.55],
+                [-869.61, 869.66],
+                [-869.62, 869.67],
+                [-869.63, 869.64],
+                [-869.64, 869.61],
+                [-869.65, 869.62],
+                [-869.66, 869.68],
+                [-869.67, 869.63],
+                [-869.68, 869.65],
+                [875.1, 875.01],
+                [876.1, 876.01],
+                [877.1, 877.01],
+                [888.1, 888.01],
+                [889.1, 889.01],
+                [892.1, 892.01],
+                [893.1, 893.01],
+                [898.1, 898.01],
+                [898.2, 898.02],
+            ];
+
+            consistentIDs.forEach(([oldID, newID]) => {
+                const pokemon = saveData.party.caughtPokemon.find(p => p.id === oldID);
+                // If player hasn't caught this mon yet, return.
+                if (pokemon == undefined) {
+                    return;
+                }
+                // Update our ID
+                pokemon.id = newID;
+                if (!saveData.statistics.pokemonHatched) {
+                    saveData.statistics.pokemonHatched = {};
+                }
+                if (!saveData.statistics.shinyPokemonHatched) {
+                    saveData.statistics.shinyPokemonHatched = {};
+                }
+                // Update our statistics
+                saveData.statistics.pokemonEncountered[newID] = saveData.statistics.pokemonEncountered[oldID] || 0;
+                saveData.statistics.pokemonDefeated[newID] = saveData.statistics.pokemonDefeated[oldID] || 0;
+                saveData.statistics.pokemonCaptured[newID] = saveData.statistics.pokemonCaptured[oldID] || 0;
+                saveData.statistics.pokemonHatched[newID] = saveData.statistics.pokemonHatched[oldID] || 0;
+                saveData.statistics.shinyPokemonEncountered[newID] = saveData.statistics.shinyPokemonEncountered[oldID] || 0;
+                saveData.statistics.shinyPokemonDefeated[newID] = saveData.statistics.shinyPokemonDefeated[oldID] || 0;
+                saveData.statistics.shinyPokemonCaptured[newID] = saveData.statistics.shinyPokemonCaptured[oldID] || 0;
+                saveData.statistics.shinyPokemonHatched[newID] = saveData.statistics.shinyPokemonHatched[oldID] || 0;
```

Comment:
Needs the male/female variants of all of these too

Reply:
Ah, of course. I believe that should be all of them?